### DISCUSSION TITLE
On object preSave, is there any way to ask user confirmation?

### DISCUSSION THREAD

**Discussion by kumar0802vivek** [pimcore/pimcore#10229] [Q&A / Support] [answered]
Hi All,

Is there any way for me to show confirmation dialog box on object preSave event? If user accept it, we have to run some deletion script.

Please suggest me. 

Thanks,

**Comment by dvesh3** [comment, marked as answer]
@kumar0802vivek just use `preSaveObject` js event
```js
    preSaveObject: function (object, type) {
        var userAnswer = confirm("Are you sure you want to save " + object.data.general.o_className + "?");
        if (!userAnswer) {
            throw new pimcore.error.ActionCancelledException('Cancelled by user');
        }
    }
``` 
https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Bundle_Developers_Guide/Plugin_Backend_UI.html#page_JavaScript-UI-Events

**Reply by kumar0802vivek** [reply]
Thanks @dvesh3.
